provider/netperf.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import os
+
+from virttest import data_dir
+from virttest import utils_misc
+from virttest import utils_net
+from virttest import utils_netperf
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class NetperfTest(object):
+
+    def __init__(self, params, vm):
+        self.params = params
+        self.vm = vm
+        self.client = self.netperf_client()
+        self.server = self.netperf_server()
+
+    def netperf_client(self):
+        """
+        Init netperf client setup
+        """
+        self.client = utils_netperf.NetperfClient(
+            self.vm.get_address(),
+            self.params.get("client_path"),
+            netperf_source=os.path.join(data_dir.get_deps_dir("netperf"),
+                                        self.params.get("netperf_client_link")),
+            client=self.params.get("shell_client"),
+            port=self.params.get("shell_port"),
+            username=self.params.get("username"),
+            password=self.params.get("password"),
+            prompt=self.params.get("shell_prompt"),
+            linesep=self.params.get("shell_linesep", "\n").encode().decode(
+                'unicode_escape'),
+            status_test_command=self.params.get("status_test_command", ""),
+            compile_option=self.params.get("compile_option", ""))
+        return self.client
+
+    def netperf_server(self):
+        """
+        Init netperf server setup
+        """
+        self.server = utils_netperf.NetperfServer(
+            utils_net.get_host_ip_address(self.params),
```

Comment:
I will recommend to make it more general, instead of fixing to host, it would be better to pass the desired server address into the class, it could be NULL and then we default it to host.